src/index.js

```diff
@@ -49,9 +50,13 @@ class CID {
    * new CID(<cid.buffer>)
    * new CID(<multihash>)
    * new CID(<bs58 encoded multihash>)
+   * new CID(<CID>)
    *
    */
   constructor (version, codec, multihash) {
+    if (CID.isCID(version)) {
+      return version
```

Comment:
My point is that you should return a reference to the same object when you call a `new`. See:

```
const cid1 = new CID(...)
const cid2 = new CID(cid1)
cid1.equals(cid2) // should return true
cid1 === cid2 // should return false
```